src/rgw/rgw_op.cc

```diff
@@ -5580,6 +5580,18 @@ int RGWBulkUploadOp::handle_dir_verify_permission()
   return 0;
 }
 
+void RGWBulkUploadOp::forward_req_info(req_info& info, std::string bucket_name)
+{
+  if (info.script_uri.find(bucket_name) != std::string::npos) {
```

Comment:
in what cases would this already contain the bucket name? maybe add a comment here to clarify?